src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -62,13 +71,28 @@
         this.transaction = trnsct;
     }
 
-    // @todo #15:30min Implement id() by parsing the string representation
-    //  of transaction according to the pattern, described in the white
-    //  paper. Replace relevant test case with actual tests.
     @Override
     @SuppressWarnings("PMD.ShortMethodName")
-    public long id() {
-        throw new UnsupportedOperationException("id() not yet implemented");
+    public int id() throws IOException {
+        final String ident = new UncheckedText(
+            new IoCheckedScalar<>(
+                new ItemAt<>(
+                    0, new SplitText(this.transaction, ";")
+                )
+            ).value()
+        ).asString();
+        if (!RtTransaction.IDENT.matcher(ident).matches()) {
+            throw new IOException(
+                new UncheckedText(
+                    new FormattedText(
+                        // @checkstyle LineLength (1 line)
+                        "Invalid ID '%s' expecting 64-bit signed hex string with 4 symbols",
```

Comment:
@carlosmiranda I just realised it should be 16-bit, not 64 here.

Reply:
@victornoel it should be _unsigned_ too, so I made two mistakes here. ;)

Reply:
@carlosmiranda hehe, I think we are all tired of working on this PR ;)